lab/src/mapreduce/common_map.go

```diff
@@ -53,6 +58,42 @@ func doMap(
 	//
 	// Your code here (Part I).
 	//
+	fileByte, err := ioutil.ReadFile(inFile)
+	checkError(err)
+
+	textContent := string(fileByte)
+	mapResults := mapF(inFile, textContent)
+
+	splitResults := make(map[int][]KeyValue)
+
+	for _, keyValuePair := range mapResults {
+		reduceIndex := int(math.Mod(float64(ihash(keyValuePair.Key)), float64(nReduce)))
+		splitResults[reduceIndex] = append(splitResults[reduceIndex], keyValuePair)
+	}
+
+	for reduceIndex, intermediateResults := range splitResults {
+		writeToIntermediateFile(reduceName(jobName, mapTask, reduceIndex), intermediateResults)
```

Comment:
Clean API!

Reply:
@Shauro you mean my API is clean or I should clear the API ? (smile-cry)